wqflask/wqflask/wgcna/gn3_wgcna.py

```diff
@@ -0,0 +1,102 @@
+"""module contains code to consume gn3-wgcna api
+and process data to be rendered by datatables
+"""
+
+import requests
+from types import SimpleNamespace
+from utility.helper_functions import get_trait_db_obs
+
+
+def fetch_trait_data(requestform):
+    """fetch trait data"""
+    db_obj = SimpleNamespace()
+    get_trait_db_obs(db_obj,
+                     [trait.strip()
+                      for trait in requestform['trait_list'].split(',')])
+
+    return process_dataset(db_obj.trait_list)
+
+
+def process_dataset(trait_list):
+    """process datasets and strains"""
+
+    input_data = {}
+    traits = []
+    strains = []
+
+    # xtodo unique traits and strains
+
+    for trait in trait_list:
+        traits.append(trait[0].name)
```

Comment:
**nitpick:** You are passing in a list, which is mutable. ATM, there's no problem here since you aren't really doing any mutable operation on it. However, if you did, this could bring some nasty bugs.

Consider:

```
from typing import List
my_list = [1, 2, 3]

def update_list(l: List):
    l.append(4)
    print("Done appending my list!")

print(my_list)
update_list(my_list)
print(my_list)
```

which outputs:

```
: [1, 2, 3]
: Done appending my list!
: [1, 2, 3, 4]
```

Notice how it modified that original list?

So if you can, best to use immutable types as args. If not, and you need to modify the args, you create a copy. With the nature of data we work with, that could be expensive. Something to think about. Cc @zsloan